diesel_tests/tests/associations.rs

```diff
@@ -187,3 +187,70 @@ fn conn_with_test_data() -> (TestConnection, User, User, User) {
 
     (connection, sean, tess, jim)
 }
+
+
+#[test]
+#[cfg(not(feature="mysql"))] // FIXME: Figure out how to handle tests that modify schema
+fn custom_foreign_key() {
+    use diesel::*;
+    use diesel::connection::SimpleConnection;
+    let connection = connection();
+    connection.batch_execute(r#"
+            DROP TABLE posts;
+            DROP TABLE users;
+            CREATE TABLE users (id SERIAL PRIMARY KEY,
+                                name TEXT NOT NULL);
+            CREATE TABLE posts (id SERIAL PRIMARY KEY,
+                                belongs_to_user INTEGER NOT NULL,
+                                title TEXT NOT NULL);
+            INSERT INTO users (id, name) VALUES (1, 'Sean'), (2, 'Tess');
+            INSERT INTO posts (id, belongs_to_user, title) VALUES
+                   (1, 1, 'Hello'),
+                   (2, 2, 'World'),
+                   (3, 1, 'Hello 2');
+        "#).unwrap();
+    table! {
+        users {
+            id -> Integer,
+            name -> Text,
+        }
+    }
+
+    table! {
+        posts {
+            id -> Integer,
+            belongs_to_user -> Integer,
+            title -> Text,
+        }
+    }
+
+    #[derive(Debug, Queryable, Identifiable, PartialEq, Clone)]
```

Comment:
Can you put diesel's derives last? Or… sort them alphabetically? Or anything that I can identify as 'order'? Or reveal the system you ordered them by? :D Sorry for commenting on all the nits